Fix -Wincompatible-pointer-type error on struct sockaddr_un with gcc 14.

--- bolt-0.9.8-origin/tests/bolt-test.c
+++ bolt-0.9.8/tests/bolt-test.c
@@ -128,7 +128,7 @@
     + strlen (sau.sun_path)
     + 1;
 
-  r = bind (fd, &sau, socklen);
+  r = bind (fd, (struct sockaddr *)&sau, socklen);
   g_assert_cmpint (r, >, -1);
 
   r = setsockopt (fd, SOL_SOCKET, SO_PASSCRED, &one, sizeof (one));
